Sources/OpenCombine/Publishers/Publishers.Map.swift

```diff
@@ -32,70 +43,120 @@ extension Publishers {
         /// The closure that transforms elements from the upstream publisher.
         public let transform: (Upstream.Output) -> Output
     }
+
+    /// A publisher that transforms all elements from the upstream publisher with a provided error-throwing closure.
+    public struct TryMap<Upstream: Publisher, Output>: Publisher {
+
+        public typealias Failure = Error
+
+        /// The publisher from which this publisher receives elements.
+        public let upstream: Upstream
+
+        /// The error-throwing closure that transforms elements from the upstream publisher.
+        public let transform: (Upstream.Output) throws -> Output
+    }
 }
 
 extension Publishers.Map {
-    /// This function is called to attach the specified `Subscriber` to this `Publisher` by `subscribe(_:)`
-    ///
-    /// - SeeAlso: `subscribe(_:)`
-    /// - Parameters:
-    ///     - subscriber: The subscriber to attach to this `Publisher`.
-    ///                   once attached it can begin to receive values.
+
     public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
-        where Output == SubscriberType.Input,
-        Upstream.Failure == SubscriberType.Failure {
-            let inner = Inner<Upstream, SubscriberType>(
-                downstream: subscriber, transform: transform
-            )
-            upstream.receive(subscriber: inner)
+        where Output == SubscriberType.Input, SubscriberType.Failure == Upstream.Failure
+    {
+        let inner = Inner(downstream: subscriber, transform: catching(transform))
+        upstream.receive(subscriber: inner)
     }
 }
 
-private final class Inner<Upstream: Publisher, Downstream: Subscriber>
-    : Subscriber,
-      Subscription {
+extension Publishers.TryMap {
+
+    public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
+        where Output == SubscriberType.Input, SubscriberType.Failure == Error
+    {
+        let inner = Inner(downstream: subscriber, transform: catching(transform))
+        upstream.receive(subscriber: inner)
+    }
+}
+
+private class _Map<Upstream: Publisher, Downstream: Subscriber>
+    : OperatorSubscription<Downstream>,
+      CustomStringConvertible,
```

Comment:
What's the reason of moving CustomStringConvertible into class instead of leaving it in the extension?

Reply:
I have a habit of implementing protocol requirements inside a class body instead of an extension if the class is non-final — this leaves the opportunity to override those methods if we need. Yep, in this case we probably won't need to override 'em, but this is a matter of consistency.